### ISSUE TITLE
Request help

### ISSUE THREAD

**Issue by ronyn77** [ollo69/ha-samsungtv-smart#318]
Hello @ollo69,

As I mentioned before, I had problems after updating the firmware of my TV, causing some audio options to disappear. Now, after a further update, the ability to choose between different HDMI ports has also disappeared within your integration. I am on firmware version 1420, the latest one.

While reviewing your documentation, I tried to add the HDMI sources manually in the configuration.yaml file, but even after restarting Home Assistant without issues, it shows me that this option is no longer used and asks me to remove it from configuration.yaml. Another thing that seems strange to me is that there are two media player entities for the TV: one created by the Samsung integration and the other by yours (which might be normal, I do not know).

Please refer to the attached picture for reference. I would appreciate any hints you can provide. Just for reference in the smart thing app on my phone, the sources are present.

Thank you.
![Snag_7fe092](https://github.com/ollo69/ha-samsungtv-smart/assets/22024600/8f97fa46-1273-44b7-9be2-c46fba3e2b7f)
![image](https://github.com/ollo69/ha-samsungtv-smart/assets/22024600/e1f9bfe8-6328-44c5-98ac-8ad24cdf970a)
![image](https://github.com/ollo69/ha-samsungtv-smart/assets/22024600/09eb7912-51a8-415f-b816-efe05ca267d7)

**Comment by LakeHouse47025** [comment]
Not sure if this helps but I have had luck calling a script to change to HDMI ports. I did not pull them into HA manually but changed the name on the TV (HDMI 2 is now CCTV for example).

service: media_player.select_source
target:
  device_id: ***YOUR DEVICE ID**
data:
  source: CCTV

**Comment by ronyn77** [comment]
I found the solution. I deleted all the integrations related to the TV, including SmartThings. I regenerated the token from scratch, then I installed it on SmartThings. Afterwards, I installed the TV with the @ollo69 integration again, using the newly generated token, and both the HDMI and audio options returned. So, it was probably the TV firmware update that caused all these problems, which now seem to be solved.
